core/arch/arm/mm/core_mmu_lpae.c

```diff
@@ -339,41 +364,96 @@ static paddr_t get_nsec_ddr_max_pa(void)
 	return pa;
 }
 
-void core_init_mmu_tables(struct tee_mmap_region *mm)
+#ifdef CFG_VIRTUALIZATION
+size_t core_mmu_get_total_pages_size(void)
 {
-	uint64_t max_va = 0;
-	size_t n;
+	return ROUNDUP(sizeof(l1_xlation_table), SMALL_PAGE_SIZE) +
+		sizeof(xlat_tables) + sizeof(xlat_tables_ul1);
+}
 
-#ifdef CFG_CORE_UNMAP_CORE_AT_EL0
-	COMPILE_TIME_ASSERT(CORE_MMU_L1_TBL_OFFSET ==
-			   sizeof(l1_xlation_table) / 2);
+struct mmu_context *core_alloc_mmu_ctx(void *tables)
+{
+	struct mmu_context *ctx;
+	char *tbl = tables;
+	unsigned int asid = asid_alloc();
+
+	assert(((vaddr_t)tbl) % SMALL_PAGE_SIZE == 0);
+
+	if (!asid)
+		return NULL;
+
+	ctx = kmalloc(sizeof(*ctx));
+
+	if (!ctx)
+		return NULL;
+
+	ctx->l1_tables = (l1_xlat_tbls_t *)(void *)tbl;
```

Comment:
Casting via `void *` isn't needed

Reply:
gcc complaints about alignment:

> warning: cast increases required alignment of target type [-Wcast-align]
>  prtn->l1_tables = (l1_xlat_tbls_t *)tbl;

I think, it can't inferre real alignment there.

Reply:
In that case drop the `(l1_xlat_tbls_t *)` cast instead.